### PR TITLE
Adding a common client module and class.

### PR DESCRIPTION
**Pull request by abonas**
Refactor the gem to have common functionality extracted to a separate module/class.

### CODE REVIEW

**Comment by abonas** [comment]
@Fryguy / @simon3z pls review. I left the kube client exception untouched in this patch,  I can generalize it too in another patch / make specific exception for each gem using this one.

File: lib/kubeclient/common.rb [hunk 1]
```diff
@@ -0,0 +1,188 @@
+require 'json'
+require 'rest-client'
+
+module Common
```
Comment (Fryguy) [inline lib/kubeclient/common.rb]: This should be futher wrapped in a Kubeclient module: `Kubeclient:Common`

Comment (abonas) [inline lib/kubeclient/common.rb]: > This should be futher wrapped in a Kubeclient module: Kubeclient:Common
@Fryguy done, any other comments or I shall merge?

**Comment by Fryguy** [comment]
I'll take a look soon, and let you know...need to wrap my head on the changes :smile:

**Comment by abonas** [comment]
> I'll take a look soon, and let you know...need to wrap my head on the changes :smile:

going once, going twice... ;)

**Comment by Fryguy** [comment]
Haha sorry... End of sprint and all :)

**Pull request merged**